dotnet/RefererParser/Parser.cs

```diff
@@ -72,20 +115,55 @@ public static Referer Parse(Uri refererUri, string pageHost = "")
                 }
             }
         }
+        
+        private Tuple<T,T,T> MakeMedia(RefererMedium @internal, RefererMedium search, RefererMedium unknown)
+        {
+            T convertedInternal;
+            T convertedSearch;
+            T convertedUnknown;
+            var internalSuccess = TryMakeMedium(@internal, out convertedInternal);
+            var searchSuccess = TryMakeMedium(search, out convertedSearch);
+            var unknownSuccess = TryMakeMedium(unknown, out convertedUnknown);
+            if (internalSuccess && searchSuccess && unknownSuccess)
+            {
+                return new Tuple<T, T, T>(convertedInternal, convertedSearch, convertedUnknown);
+            }
+
+            var errorString = new StringBuilder("Supplied enum missing values for: ");
+            if (internalSuccess)
```

Comment:
shouldn't this be `if (!internalSuccess)`?

Reply:
You're absolutely right, and had I written the correct tests, I would've caught that. Thanks.